compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -269,7 +269,8 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                     mir::CastKind::Pointer(
                         PointerCast::MutToConstPointer | PointerCast::ArrayToPointer,
                     )
-                    | mir::CastKind::Misc => {
+                    | mir::CastKind::Misc
+                    | mir::CastKind::PointerFromExposedAddress => {
```

Comment:
Since this can have arbitrary integer types as input, it is currently best handled in the same code path as the other integer-to-X casts. Miri shows that this could be factored differently but for now this seems easier.

Reply:
maybe leave this as a comment on the source?

Reply:
Fair, done.